src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/9.diff.sql

```diff
@@ -0,0 +1,441 @@
+
+/*************************************************************
+    Task Table
+**************************************************************/
+IF NOT EXISTS (SELECT 'X' FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 'TaskInfo')
+BEGIN
+    CREATE TABLE [dbo].[TaskInfo](
+	    [TaskId] [varchar](64) NOT NULL,
+	    [QueueId] [varchar](64) NOT NULL,
+	    [Status] [smallint] NOT NULL,
+        [TaskTypeId] [smallint] NOT NULL,
+        [RunId] [varchar](50) null,
+	    [IsCanceled] [bit] NOT NULL,
+        [RetryCount] [smallint] NOT NULL,
+	    [HeartbeatDateTime] [datetime2](7) NULL,
+	    [InputData] [varchar](max) NOT NULL,
+	    [TaskContext] [varchar](max) NULL,
+        [Result] [varchar](max) NULL
+    ) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
+END 
+
+IF NOT EXISTS (SELECT 'X' FROM SYS.INDEXES WHERE name = 'IXC_Task' AND OBJECT_ID = OBJECT_ID('TaskInfo'))
+BEGIN
+    CREATE UNIQUE CLUSTERED INDEX IXC_Task on [dbo].[TaskInfo]
+    (
+        TaskId
+    )
+END
+GO
+
+/*************************************************************
+    Stored procedures for general task
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     CreateTask
+--
+-- DESCRIPTION
+--     Create task for given task payload.
+--
+-- PARAMETERS
+--     @taskId
+--         * The ID of the task record to create
+--     @queueId
+--         * The number of seconds that must pass before an export job is considered stale
+--     @taskTypeId
+--         * The maximum number of running jobs we can have at once
+--
+CREATE OR ALTER PROCEDURE [dbo].[CreateTask]
+    @taskId varchar(64),
+    @queueId varchar(64),
+	@taskTypeId smallint,
+    @inputData varchar(max)
+AS
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @heartbeatDateTime datetime2(7) = SYSUTCDATETIME()
+	DECLARE @status smallint = 1
+    DECLARE @retryCount smallint = 0
+	DECLARE @isCanceled bit = 0
+
+    -- Check if the task already be created
+    DECLARE @existedTaskCount bigint
+
+    SELECT @existedTaskCount = Count(*)
+    FROM [dbo].[TaskInfo]
+    WHERE TaskId = @taskId
```

Comment:
Is this necessary since we have another check below?

Reply:
Yes, it is duplicated.